presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcPageSourceFactory.java

```diff
@@ -206,6 +207,9 @@ public static OrcPageSource createOrcPageSource(
             if (e.getClass().getSimpleName().equals("BlockMissingException")) {
                 throw new PrestoException(HIVE_MISSING_DATA, message, e);
             }
+            else if (e.getClass().getSimpleName().equals("InvalidProtocolBufferException")) {
```

Comment:
Oh.  The problem is the constructor is not using the metadata reader that is wrapped in a `ExceptionWrappingMetadataReader`.  Lets fix that and remove this check.